src/libs/Navigation/AppNavigator/useNavigationResetRootOnLayoutChange.ts

```diff
@@ -9,13 +10,14 @@ import navigationRef from '@libs/Navigation/navigationRef';
  */
 function useNavigationResetRootOnLayoutChange() {
     const {shouldUseNarrowLayout} = useResponsiveLayout();
+    const prevShouldUseNarrowLayout = usePrevious(shouldUseNarrowLayout);
 
     useEffect(() => {
-        if (!navigationRef.isReady()) {
+        if (!navigationRef.isReady() || shouldUseNarrowLayout === prevShouldUseNarrowLayout) {
             return;
         }
         navigationRef.resetRoot(navigationRef.getRootState());
-    }, [shouldUseNarrowLayout]);
+    }, [prevShouldUseNarrowLayout, shouldUseNarrowLayout]);
```

Comment:
I feel like we do not need both as they are technically always intertwined. If one changes the other one has to change too. If I have removed this, it works 
```suggestion
    }, [shouldUseNarrowLayout]);
```